### ISSUE TITLE
Custom Front Matter slugs using Gatsby Page Creator

### ISSUE THREAD

**Issue by zhandosweb** [gatsbyjs/gatsby#21395]
Hello!

Please, let me ask one question about Gatsby Page Creator plugin. https://www.gatsbyjs.org/packages/gatsby-plugin-page-creator/

If I'm using this plugin to create pages for MDX posts. Can I set slugs for my MDX-posts using permalink in front matter? 

For example, I write into Front Matter something like this: 

```
---
permalink: /custom-url/
---
```

How I can set permalink (slug) written into Front Matter for Gatsby Page Creator? In plugins documentation, I no found any information. Sorry for the noob question.

**Comment by wardpeet** [comment]
Hey, the gatsby-plugin-page-creator does not give you the option to do so.

You can do a similar thing in gatsby-node.js. More info can be found at https://www.gatsbyjs.org/docs/adding-markdown-pages/, you have to change markdown into mdx for your use case.

```
const path = require(`path`)
exports.createPages = async ({ actions, graphql, reporter }) => {
  const { createPage } = actions
  const blogPostTemplate = path.resolve(`src/templates/blogTemplate.js`)
  const result = await graphql(`
    {
      allMdx(
        sort: { order: DESC, fields: [frontmatter___date] }
        limit: 1000
      ) {
        nodes {
          frontmatter {
            permalink
          }
        }
      }
    }
  `)
  // Handle errors
  if (result.errors) {
    reporter.panicOnBuild(`Error while running GraphQL query.`)
    return
  }
  result.data.allMdx.nodes.forEach(({ node }) => {
    createPage({
      path: node.frontmatter.permalink,
      component: blogPostTemplate,
      context: {}, // additional data can be passed via context
    })
  })
}
```

I'm marking this issue as answered and closing it for now but please feel free to reopen this and comment if you would like to continue this discussion. We hope we managed to help and thank you for using Gatsby! 💜

**Comment by zhandosweb** [comment]
Thank you very much for your answer!

Yes, I know about gatsby-node.js. But it will be very nice, if we can set custom slugs and post templates using Gatsby Page Creator plugin. 

And there Gatsby will be easy like Wordpress. And any noob like me can easily work with Gatsby))

Thank you again for your answer! You do the web better with Gatsby!

**Comment by zhandosweb** [comment]
Sorry sir, and let me one question again. 

Can I set a post template file using Gatsby Page Creator plugin? 

For example, in gatsby-node.js we set Blog Post template like this: `const blogPostTemplate = path.resolve(`src/templates/blogTemplate.js`)`

Can I set Blog Post Template using Gatsby Page Creator plugin? Or my MDX-posts using no layouts? (empty template, and only clean text in browser XD)

Sorry for bad English.

**Comment by zhandosweb** [comment]
Sorry sir, and let me ask one question again. 

Can I set default posts Layout using Gatsby Page Creator plugin? Or my MDX-posts don't have a layout, and in-browser I can see the only plain text? 

For example, in gatsby-node.js we set default Blog Post Template with the command: `const blogPostTemplate = path.resolve(`src/templates/blogTemplate.js`)`

Can I do something with Gatsby Page Creator plugin?
